.github/workflows/publish-docs.yaml

```
@@ -0,0 +1,45 @@
+---
+name: publish documentation
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Install Poetry
+        run: |
+          pip install poetry
+      - name: Install Dependencies
+        run: |
+          poetry install
```

Once my [suggestion applied](https://github.com/scality/gh-actions-exporter/pull/18/files#r1174026094)
```suggestion
          poetry install --only docs
```